### PR TITLE
[WIP] Implement signature verification

### PR DESCRIPTION
**Pull request by MurashovVen**
Implementation of signature authentication for operations with allocations.
Need to be merged with additional changes on gosdk client.

### CODE REVIEW

**Comment by MurashovVen** [comment]
Verifying the signature by the owner's public key should solve the problem of a fake client id. 
If we decide to remove the Client Key and Client ID headers, I'll make the appropriate changes.

**Review by Sriep — COMMENTED** [review]
Currently blobber is not unit test friendly. The best way to mock the database needs to be considered.

File: code/go/0chain.net/blobbercore/handler/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,1064 @@
...
+			{
+				ID:           1,
+				AllocationID: allocID,
+			},
+		},
+		Expiration: common.Timestamp(time.Now().Add(time.Hour).Unix()),
+	}
+	return &alloc
+}
+
```
[the remark this answers is not in the document]
Comment (Sriep) [reply code/go/0chain.net/blobbercore/handler/handler_test.go:103]: PR82 replaced with RP https://github.com/0chain/blobber/pull/87 which is a version of `setupDB` moved to `datastore` for better visibility. It's likely that most blobber unit tests will need to mock the Postgres DB.

[the remark this answers is not in the document]
Comment (andrenerd) [reply code/go/0chain.net/blobbercore/handler/handler_test.go]: @Sriep 
could we merge it, as the front end guys need new features implemented by this update.
and concerning the tests, i'm assigned a task to create more tests for blobber. so i will care about mocking too. thanks.

**Comment by stewartie4** [comment]
What's the status of this PR and https://github.com/0chain/gosdk/pull/30 ?

**Review by platsko — APPROVED** [review]

**Pull request merged**